hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -139,10 +142,17 @@ public static DataStreamSink<Object> bulkInsert(Configuration conf, RowType rowT
         dataStream = dataStream.partitionCustom(partitioner, rowDataKeyGen::getPartitionPath);
       }
       if (conf.getBoolean(FlinkOptions.WRITE_BULK_INSERT_SORT_INPUT)) {
-        SortOperatorGen sortOperatorGen = new SortOperatorGen(rowType, partitionFields);
-        // sort by partition keys
+        String[] sortFields = partitionFields;
+        if (conf.getBoolean(FlinkOptions.WRITE_BULK_INSERT_SORT_INPUT_BY_RECORD_KEY)) {
+          // sort by record keys
+          ArrayList<String> sortList = new ArrayList<>(Arrays.asList(partitionFields));
+          Collections.addAll(sortList, conf.getString(FlinkOptions.RECORD_KEY_FIELD).split(","));
+          sortFields = sortList.toArray(new String[0]);
+        }
+        SortOperatorGen sortOperatorGen = new SortOperatorGen(rowType, sortFields);
+        // sort by partition keys or (partition keys and record keys)
         dataStream = dataStream
-            .transform("partition_key_sorter",
+            .transform("partition_and_record_key_sorter",
                 InternalTypeInfo.of(rowType),
                 sortOperatorGen.createSortOperator())
```

Comment:
Generate operator name as `sorter:(partition_key)` or `sorter:(partition_key, record_key)` based on the given configurations.

Reply:
Have given you the example operator above.